### PR TITLE
Include msg of CannotParse failure into its description.

### PR DESCRIPTION
**Pull request by jbgi**

### CODE REVIEW

**Review by jcazevedo — CHANGES_REQUESTED** [review]
Thanks for your PR! I agree it makes sense to include the message of a `CannotParse` failure into its description.

Unfortunately, this way we end up including the location of the failure twice, as the `ConfigException.Parse` exception message includes it and we also include it when we list the failures in the `ConfigReaderException` message. Maybe we can strip the prefix of the `ConfigOrigin`'s description from the `ConfigException.Parse` message (see [ConfigException.java#L23](https://github.com/typesafehub/config/blob/master/config/src/main/java/com/typesafe/config/ConfigException.java#L23)).

Please also update the tests in the `ConfigReaderExceptionSuite` accordingly.

**Comment by melrief** [comment]
@jbgi thanks for the PR! I agree with @jcazevedo about the changes that should be done to get this merged in master. Both the breaking test and the double inclusion of the location should be addressed before we continue.

**Comment by jbgi** [comment]
@jcazevedo, @melrief PR updated with requested changes.

File: core/src/main/scala/pureconfig/backend/ConfigFactoryWrapper.scala
Comment (melrief) [inline core/src/main/scala/pureconfig/backend/ConfigFactoryWrapper.scala:42]: I'll link the source code of [`ConfigException`](https://github.com/typesafehub/config/blob/master/config/src/main/java/com/typesafe/config/ConfigException.java#L21) in case somebody else wants to check why we need do strip the prefix in this way. `ConfigException` takes in the constructor origin and [appends the description to the exception message](https://github.com/typesafehub/config/blob/master/config/src/main/java/com/typesafe/config/ConfigException.java#L23).

**Review by melrief — APPROVED** [review]
LGTM @jbgi, thanks for doing this. I'll wait for another review and then merge.

**Review by jcazevedo — APPROVED** [review]

**Comment by jcazevedo** [comment]
Thanks for the PR @jbgi! We would like to add you to the [CONTRIBUTORS](https://github.com/pureconfig/pureconfig/blob/master/CONTRIBUTORS) file. If you agree, please submit another PR in which you add a line with your name, your email (optional) and your username, in line with PRs such as #118, #152 or #160.

**Comment by jbgi** [comment]
Thanks @jcazevedo, however I am fine with only the git log attribution.

**Pull request merged**
